### ISSUE TITLE
`error: module not found: com.kohlschutter.annotations.compiletime`

### ISSUE THREAD

**Issue by saksmt** [kohlschutter/junixsocket#148]
Getting compile error with 2.8.3, gradle 8.2 and JDK JBR 17.0.7-b829.16:
```
error: module not found: com.kohlschutter.annotations.compiletime
```

Dependencies in gradle:
```gradle
implementation("com.kohlschutter.junixsocket:junixsocket-core:2.8.3")
```

`module-info.java` of project:
```java
module myModule {
    requires java.base;
    requires org.newsclub.net.unix;
}
```

**Notes**

I assume problem is in module-info of junixsocket-common, specifically [`requires static transient` part](https://github.com/kohlschutter/junixsocket/blob/junixsocket-2.8.3/junixsocket-common/src/main/java/module-info.java#L11C1-L11C71) - it should not be transient, since transient propagates `requires static` to all modules using it.

**Comment by kohlschuetter** [comment]
Thanks for reporting, @saksmt!

Please verify with the latest `2.9.0-SNAPSHOT` build.

For testing snapshot builds, please specify the following section in your `build.gradle` file:
```
repositories {
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
        mavenContent {
            snapshotsOnly()
        }
    }
}
```

**Comment by saksmt** [comment]
@kohlschuetter Works like a charm! Thanks! But I seems like my assumption was wrong (since that `require static transitive` is still present), can you share what was the issue?

**Comment by kohlschuetter** [comment]
By the way, if you can't wait for the 2.9.0 release, add this dependency to your project:

```
implementation("com.kohlschutter:compiler-annotations:1.6.4")
```

**Comment by kohlschuetter** [comment]
Fixed in 2.9.0, closing.
